File: Leetcode/PY/Leetcode372-超级次方.py
class Solution:
    def superPow(self, a: int, b: List[int]) -> int:
        p = 1337        
        def qmi(a,b):
            a %= p
            res = 1
            while b != 0:
                if b & 1 == 1: res = res * a % p 
                b >>= 1
                a = a * a % p
            return res
        if not b: return 1
        k = b[-1]
        b.pop()
        return qmi(self.superPow(a,b),10) * qmi(a,k) % p

File: Leetcode/JS/Leetcode372-超级次方.js
var superPow = function(a, b) {
    let p = 1337
    let qmi = function(a,b){
        a %= p;
        let res = 1;
        while(b) {
            if(b & 1) res = res * a % p;
            b >>= 1;
            a = a * a % p;
        }
        return res;
    }
    if(b.length == 0) return 1;
    let k = b.pop();
    return qmi(superPow(a,b),10) * qmi(a,k) % p;
};